src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1044,39 +1044,47 @@ void CodeGen::genHWIntrinsic_R_R_R_RM(
 
             regSet.tmpRlsTemp(tmpDsc);
         }
-        else if (op3->OperIsHWIntrinsic())
+        else if (op3->isIndir() || op3->OperIsHWIntrinsic())
```

Comment:
Are similar fixes needed for all the other `genHWIntrinsic_` code? Such as: https://github.com/dotnet/coreclr/blob/4ff4ef182e148b8924853aca3d49b80aa9cf926a/src/jit/hwintrinsiccodegenxarch.cpp#L435

Reply:
I suspect so. However, I will file a separate issue that includes extending the testing to include more containment cases.

Reply:
If it is just this same thing on all of them, I can address that and you can feel free to assign the issue to me.

Reply:
Right - it would be just the same, but I don't want to introduce the code without a test to verify it (plus I thought it would be prudent to do it separately, to keep this focused on fixing the existing failures).